Java/jni/native_view.cc

```diff
@@ -149,24 +149,26 @@ JNIEXPORT void JNICALL Java_com_couchbase_cbforest_View_emit(JNIEnv *env, jobjec
     jlong *keys   = env->GetLongArrayElements(jkeys, NULL);
     C4Key *c4keys[count];
     C4Slice c4values[count];
-    std::vector<jbyteArraySlice> valueBufs;
+    std::vector<jbyteArraySlice*> valueBufs;
     for(int i = 0; i < count; i++) {
         c4keys[i] = (C4Key*)keys[i];
         jbyteArray jvalue = (jbyteArray) env->GetObjectArrayElement(jvalues, i);
         if (jvalue) {
-            valueBufs.push_back(jbyteArraySlice(env, jvalue));
```

Comment:
I'm starting to remember why I hate C++. It looks like even with `emplace`, the copy constructor gets called, but the whole point of emplace is that it doesn't do that. The STL source code is unreadable, and even the docs for `emplace` don't make sense to me … it's crazy template stuff.

Another solution is to define a "move constructor" for `jbyteArray`. I'll send you a modified copy of `native_glue.hh` with that; can you try it out?

I know that your solution above would work, but I want to avoid malloc. This bit of code is going to be a bottleneck because it's part of the `emit()` call, so I want to make sure it's as fast as possible.